File: Tests/Test_IsAdmin.vb
Option Explicit On
Option Strict On
Option Compare Binary
Option Infer Off

Imports System
Imports System.IO

Namespace Tests
    Module Tests_IsAdmin
        Function Test_IsAdmin1() As Boolean
            Return TestBoolean("IsAdmin1", WalkmanLib.IsAdmin(), False)
        End Function

        Function Test_IsAdmin2(rootTestFolder As String) As Boolean
            Dim programPath As String = Reflection.Assembly.GetExecutingAssembly().CodeBase
            programPath = New Uri(programPath).LocalPath
            Dim tmpOutPath As String = Path.Combine(rootTestFolder, "outTmp.txt")

            WalkmanLib.RunAsAdmin("cmd.exe", "/c """ & programPath & """ getAdmin > " & tmpOutPath)
            Threading.Thread.Sleep(1000)

            Dim runAsAdminOutput As String
            Try
                runAsAdminOutput = File.ReadAllText(tmpOutPath)
            Catch ex As Exception
                runAsAdminOutput = "Error: " & ex.Message
            End Try
            Try
                File.Delete(tmpOutPath)
            Catch
            End Try

            Return TestString("IsAdmin2", runAsAdminOutput, "True" & Microsoft.VisualBasic.vbNewLine)
        End Function
    End Module
End Namespace

File: Tests/Test_IsAdmin.cs
using System;
using System.IO;

namespace Tests {
    static class Tests_IsAdmin {
        public static bool Test_IsAdmin1() {
            return GeneralFunctions.TestBoolean("IsAdmin1", WalkmanLib.IsAdmin(), false);
        }

        public static bool Test_IsAdmin2(string rootTestFolder) {
            string programPath = System.Reflection.Assembly.GetExecutingAssembly().CodeBase;
            programPath = new Uri(programPath).LocalPath;
            string tmpOutPath = Path.Combine(rootTestFolder, "outTmp.txt");

            bool result = WalkmanLib.RunAsAdmin("cmd.exe", "/c \"" + programPath + "\" getAdmin > " + tmpOutPath);
            if (!result)
                return GeneralFunctions.TestString("IsAdmin2", "Admin Prompt Denied", "Admin Prompt Accepted");

            string runAsAdminOutput;
            try {
                runAsAdminOutput = File.ReadAllText(tmpOutPath);
            } catch (Exception ex) {
                runAsAdminOutput = "Error: " + ex.Message;
            }
            GeneralFunctions.DeleteFileIfExists(tmpOutPath);

            return GeneralFunctions.TestString("IsAdmin2", runAsAdminOutput, "True" + Environment.NewLine);
        }
    }
}
